BaseApp/COD3051.TXT

```diff
@@ -0,0 +1,209 @@
+OBJECT Codeunit 3051 DotNet_RegexOptions
+{
+  OBJECT-PROPERTIES
+  {
+    Date=;
+    Time=;
+    Version List=;
+  }
+  PROPERTIES
+  {
+    OnRun=BEGIN
+          END;
+
+  }
+  CODE
+  {
+    VAR
+      DotNetRegexOptions@1000000001 : DotNet "'System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.System.Text.RegularExpressions.RegexOptions";
+
+    [External]
+    PROCEDURE Compiled@1000000009() : Integer;
+    BEGIN
+      DotNetRegexOptions := DotNetRegexOptions.Compiled;
+      EXIT(DotNetRegexOptions);
+    END;
+
+    [External]
+    PROCEDURE CultureInvariant@1000000017() : Integer;
+    BEGIN
+      DotNetRegexOptions := DotNetRegexOptions.CultureInvariant;
+      EXIT(DotNetRegexOptions);
+    END;
+
+    [External]
+    PROCEDURE ECMAScript@1000000018() : Integer;
+    BEGIN
+      DotNetRegexOptions := DotNetRegexOptions.ECMAScript;
+      EXIT(DotNetRegexOptions);
+    END;
+
+    [External]
+    PROCEDURE ExplicitCapture@1000000019() : Integer;
+    BEGIN
+      DotNetRegexOptions := DotNetRegexOptions.ExplicitCapture;
+      EXIT(DotNetRegexOptions);
+    END;
+
+    [External]
+    PROCEDURE IgnoreCase@1000000020() : Integer;
+    BEGIN
+      DotNetRegexOptions := DotNetRegexOptions.IgnoreCase;
+      EXIT(DotNetRegexOptions);
+    END;
+
+    [External]
+    PROCEDURE IgnorePatternWhitespace@1000000021() : Integer;
+    BEGIN
+      DotNetRegexOptions := DotNetRegexOptions.IgnorePatternWhitespace;
+      EXIT(DotNetRegexOptions);
+    END;
+
+    [External]
+    PROCEDURE Multiline@1000000022() : Integer;
+    BEGIN
+      DotNetRegexOptions := DotNetRegexOptions.Multiline;
+      EXIT(DotNetRegexOptions);
+    END;
+
+    [External]
+    PROCEDURE None@100000028() : Integer;
+    BEGIN
+      DotNetRegexOptions := DotNetRegexOptions.None;
+      EXIT(DotNetRegexOptions);
+    END;
+
+    [External]
+    PROCEDURE RightToLeft@1000000024() : Integer;
+    BEGIN
+      DotNetRegexOptions := DotNetRegexOptions.RightToLeft;
+      EXIT(DotNetRegexOptions);
+    END;
+
+    [External]
+    PROCEDURE Singleline@1000000025() : Integer;
+    BEGIN
+      DotNetRegexOptions := DotNetRegexOptions.Singleline;
+      EXIT(DotNetRegexOptions);
+    END;
+
+    [External]
+    PROCEDURE BitwiseOr@100000016(Option1@100000000 : Integer;Option2@100000001 : Integer) : Integer;
```

Comment:
Unfortunately, we cannot accept this kind of functions in a .Net wrapper.

You can define it in your own code, or use + in this case.
`RegexOptions` values are power of two so summing will have the same effect as bit-wise or.